[Feature] Qualified recruitment card

esizer opened the pull request:
🤖 Resolves #6737 

## 👋 Introduction

Adds a new component `QualifiedRecruitmentCard`.

## 🕵️ Details

Some minor adjustments have been made to the dialog (to be submitted to designers for review) to attempt to improve accessibility. Some changes that were made:

- Updated the title to include the pool title since it is a label for the dialog and should be unique
- Reduced the amount of bold text to meet[ the WebAIM readability suggestions for cognitive accessibility](https://webaim.org/articles/evaluatingcognitive/#readability:~:text=Overuse%20can%20result%20in%20the%20loss%20of%20differentiation.%20Do%20not%20use%20italics%20or%20bold%20on%20long%20sections%20of%20text.%20Be%20careful%20with%20ALL%20CAPS.)
- Similar, the word "NOT" was all uppercase which can be read incorrectly by certain ATs and makes it more difficult for users with disability such as dyslexia to read
- Button order has been swapped, this one is far more subjective but it seems (at least to me) as though "Cancel" is the least "destructive" action so it should lead. 

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Run storybook `npm run storybook`
2. Navigate to the Qualified Recruitment Card story
3. Confirm the states match [the expected states in the mockup](https://www.figma.com/file/2VF8cE4cctTBQUkPBcVxUG/3.-Patterns?type=design&node-id=703-60935&t=urWlzAtSRrR080MT-0)

## 📸 Screenshot

### Qualified - Available
![Screenshot 2023-06-13 095834](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/0f655a6e-cefe-45ce-94ec-8f47262e6f8d)

### Qualified - Not Available
![Screenshot 2023-06-13 100109](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/aeaecccd-21b0-41c4-b7b8-f368fa6d4e1a)

### Expired
![Screenshot 2023-06-13 100115](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/d30f00f7-79ea-4cf3-ac8a-9bceb121af01)

### Placed
![Screenshot 2023-06-13 100121](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/141e3826-e3e0-4a4c-8e7e-22c01f56f665)

### Open
![Screenshot 2023-06-13 100842](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/28dbfc15-abae-42ed-889b-16fcfab72151)

### Dialog 
![Screenshot 2023-06-13 100101](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/32ecfe41-50bc-48b0-a611-048169e70e71)

substrae commented:
@esizer 

- Not sure I agree with the title change; this is now super difficult to read from a sighted perspective, and is immediately redundant given the next line of text
- I've updated the designs with the reduced bolding
- I've updated the designs with the lowercased "not"
- We've taken an approach with buttons recently based on research @Jerryescandon had done around both left-aligning actions and promoting the primary action to always be first, with secondary actions after

esizer commented:
> * Not sure I agree with the title change; this is now super difficult to read from a sighted perspective, and is immediately redundant given the next line of text
> * I've updated the designs with the reduced bolding
> * I've updated the designs with the lowercased "not"
> * We've taken an approach with buttons recently based on research @Jerryescandon had done around both left-aligning actions and promoting the primary action to always be first, with secondary actions after

Thanks, I have updated the title and button order in [update title, button order](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6940/commits/cf8b0ca590d60f9f9073a4b03ad4f15bc2ebe743).

Related to the button order, we should do that sooner rather than later. I noticed we have introduced some inconsistency in the order which could cause some confusion. I'll see if we can get that planned as part of the next sprint 👍

substrae commented:
Oh one small thing I noticed, I think the title font-size on the cards is off by one level. Thanks!

esizer commented:
> Oh one small thing I noticed, I think the title font-size on the cards is off by one level. Thanks!

Hmm, if that is correct then we may have a larger issue with font size 😅 Right now it appears at `h6` (`18.4px`) since that was [how the issue was defined](https://github.com/GCTC-NTGC/gc-digital-talent/issues/6737#:~:text=The%20card%20title%20should%20use%20the%20h6%20font%2Dsize%20while%20being%20whatever%20HTML%20heading%20is%20relevant%20for%20the%20page%20context). If it is not correct then the hydrogen config may be out of sync 😬 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/91711484d7e4e5374c44d8167bab670493973d2b/apps/web/src/components/QualifiedRecruitmentCard/QualifiedRecruitmentCard.tsx#L81-L83

![Screenshot 2023-06-13 132813](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/1b99fa58-d68a-49c3-b881-7eaa91f1f20d)

vd1992 commented on apps/web/src/components/RecruitmentAvailabilityDialog/RecruitmentAvailabilityDialog.tsx:
```diff
@@ -0,0 +1,192 @@
+                })}
+                idPrefix="availability"
+                id="isSuspended"
+                name="isSuspended"
+                items={[
+                  {
+                    label: intl.formatMessage({
+                      defaultMessage:
+                        "I am <strong>available</strong> for hire and want to be contacted about opportunities from this recruitment process.",
+                      id: "cAOf3a",
+                      description:
+                        "Radio button label for available for opportunities option",
+                    }),
+                    value: "true",
```
This works, but it is a little confusing to follow as `isSuspended` is essentially flipped for the form where `true` maps to not suspended.
Connects to this bit below where `isSuspended = true` sets `isSuspended = false`
Not a big deal as it appears to work fine, maybe just a comment or renaming the second variable/form to `isActive` if you want. 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/91711484d7e4e5374c44d8167bab670493973d2b/apps/web/src/components/RecruitmentAvailabilityDialog/RecruitmentAvailabilityDialog.tsx#L32-L33

esizer replied:
Oof, yeah. That is wrong 😓 I got confused because the form is a positive statement "I am available" so I was thinking positive statement === true 😜 

Fixed in [invert isSuspended logic for availability dialog](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6940/commits/f891ecb079c5be9bab6b770e56d911ff9b22f00e)

vd1992 commented on apps/web/src/components/RecruitmentAvailabilityDialog/RecruitmentAvailabilityDialog.tsx:
```diff
@@ -0,0 +1,192 @@
+  } = methods;
+
+  const updateSuspendedAtStatus = async (values: FormValues) => {
+    await executeMutation({
+      id: candidate.id,
+      isSuspended: values.isSuspended === "true",
```
Hang on, maybe the above will pose a problem after all. 
As "I am available" maps to "true", this passes in `isSuspended = true`, so the opposite of how it should play out.
Selecting "I am available" will suspend your pool candidate.

vd1992 replied:
Yep, imported the component in to try it out

https://recordit.co/EJ1GVuk9IK

vd1992 commented on apps/web/src/components/QualifiedRecruitmentCard/QualifiedRecruitmentCard.tsx:
Might be worth it to stick margins around it so a list of them don't need a containing div to space them out?
Not too sure about this comment, so disregard if you want 👍 

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/29c0550d-7313-4f8a-a3ed-dc43a006b27f)

esizer replied:
Spacing can be determined by the layout of a page and can change depending on where a specific component is being displayed. Baking in a basic margin complicates this so it is avoided in favour of flex gap. I'm willing to listen to other opinions though if others feel we should have spacing baked into the components.

vd1992 replied:
👍

vd1992 commented on apps/web/src/messages/poolCandidateMessages.ts:
I think we are supposed to use "recruitment process" for live site copy rather than "recruitment", might need to double check at stand-up

vd1992 replied:
Non issue

vd1992 reviewed (changes requested):
Checked around some places, not exhaustive as I am poor at reviewing things with lots of moving parts. 
Found some stuff, take a look. Otherwise, looks awesome

brindasasi commented on apps/web/src/components/QualifiedRecruitmentCard/QualifiedRecruitmentCard.tsx:
```diff
@@ -0,0 +1,343 @@
+        data-h2-color="base(secondary.darker)"
+        data-h2-margin="base(x.5 0 x1 0)"
+      >
+        {isOngoing
```
oh am doing the samething .. I think we are overlapping ..will base mine of yours then.
Do you mind taking this out into a utility function something like getRecruitmentType?

esizer replied:
Yeah, of course! I can do that and let you know when it is completed 👍

brindasasi commented:
Looks like we are missing **Awarded on** 
![Screenshot 2023-06-14 at 10 27 34 AM](https://github.com/GCTC-NTGC/gc-digital-talent/assets/89482910/57ac2aaa-565b-434c-adc8-301d13d59138)

brindasasi commented:
> > Looks like we are missing **Awarded on**
> 
> The acceptance criteria states to [not display awarded on](https://github.com/GCTC-NTGC/gc-digital-talent/issues/6737#:~:text=Hide%20the%20date%20for%20now%20%2D%20we%20do%20not%20currently%20record%20the%20%22accepted%2Dat%22%20date).
oh yeah .. forgot about that. Nvm! :)

esizer commented on apps/web/src/components/RecruitmentAvailabilityDialog/RecruitmentAvailabilityDialog.tsx:
```diff
@@ -0,0 +1,192 @@
+          color="black"
+          aria-label={intl.formatMessage(
+            {
+              defaultMessage: "Change your availability for {title}",
```
Fixed in [Add period to aria-label](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6940/commits/db59c1f102f19cede6f26f14d355799c068cbd6b) so the ID would also update to match.

esizer commented:
> This is bold in figma

Good catch! 👍 Fixed in [make status pill bold](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6940/commits/e66adec870c6396ab585f8e3d354d1382eb3f730)

brindasasi reviewed (commented):
Are you doing this ? https://github.com/GCTC-NTGC/gc-digital-talent/pull/6940/files/f891ecb079c5be9bab6b770e56d911ff9b22f00e#r1229684097
Everything else looks good to me! 👍🥳

brindasasi commented:
Looks 

> Are you doing this ? https://github.com/GCTC-NTGC/gc-digital-talent/pull/6940/files/f891ecb079c5be9bab6b770e56d911ff9b22f00e#r1229684097 Everything else looks good to me! 👍🥳

oops .. just noticing ..thanks! 😅

brindasasi reviewed (commented):
minor thing: **getTargetedRecruitment**  name doesn't suit well as TargetedRecruitment is a type of recruitment and this function will also be used for ongoing recruitment as well. 

**getRecruitmentType** or something like that would be suitable

brindasasi reviewed (approved):
I think you are mind reading me 😆

Thank you and this is good to go with translations👏👏👏
